File: app/src/main/java/edu/schoolapp/TimeTableAdapter.java
package edu.schoolapp;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

public class TimeTableAdapter extends RecyclerView.Adapter {

//    List<>

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }

    static class TimeTableViewHolder extends RecyclerView.ViewHolder {

        public TimeTableViewHolder(View itemView) {
            super(itemView);
        }
    }

    static class TimeTableHeaderViewHolder extends RecyclerView.ViewHolder {

        public TimeTableHeaderViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/edu/schoolapp/TimeTableAdapter.kt
package edu.schoolapp

import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class TimeTableAdapter : RecyclerView.Adapter<TimeTableAdapter.TimeTableViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TimeTableViewHolder {
        TODO("Not yet implemented")
    }

    override fun onBindViewHolder(holder: TimeTableViewHolder, position: Int) {
        TODO("Not yet implemented")
    }

    override fun getItemCount(): Int = 0

    inner class TimeTableViewHolder(itemView: View) : RecyclerView.ViewHolder(
        itemView
    )
}